### PR TITLE
CSI-4884  add env tests

### PR DESCRIPTION
**Pull request by matancarmeli7**
related PRs:

- #39
- #40 
- #41

### CODE REVIEW

File: controllers/envtest/volumegroup_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"time"
+
+	volumegroupv1 "github.com/IBM/csi-volume-group-operator/api/v1"
+	"github.com/IBM/csi-volume-group-operator/controllers/envtest/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Controller", func() {
+	Describe("Test controllers", func() {
```
Comment (ArbelNathan) [inline controllers/envtest/volumegroup_controller_test.go:30]: indentation (if even needed the struct)

File: controllers/envtest/volumegroupcontent_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+				Expect(err).NotTo(HaveOccurred())
+
+				err = utils.CreateResourceObject(VGClass, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
+				vgclass := &volumegroupv1.VolumeGroupClass{}
+				err = utils.GetNamespacedResourceObject(VGClassName, Namespace, vgclass, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
+				vgclass.VolumeGroupDeletionPolicy = &retainDeletionPolicy
+				err = k8sClient.Update(context.TODO(), vgclass)
+				Expect(err).NotTo(HaveOccurred())
```
Comment (ArbelNathan) [inline controllers/envtest/volumegroupcontent_controller_test.go:52]: can add arg to create create resource method

File: controllers/envtest/volumegroup_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+				Expect(err).NotTo(HaveOccurred())
+
+				close(done)
+			}, Timeout.Seconds())
+			It("should add and remove volume objects from volumeGroup objects when created before vg", func(done Done) {
+				By("Creating volume objects before volumeGroup objects")
+				err := utils.CreateResourceObject(Secret, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
+				err = utils.CreateResourceObject(StorageClass, k8sClient)
+				Expect(err).NotTo(HaveOccurred())
```
Comment (ArbelNathan) [inline controllers/envtest/volumegroup_controller_test.go:64]: can we unify this two objects creation?

File: controllers/envtest/volumegroup_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+				err = createVolumeObjects()
+				Expect(err).NotTo(HaveOccurred())
+				err = createVolumeGroupObjects()
+				Expect(err).NotTo(HaveOccurred())
```
Comment (ArbelNathan) [inline controllers/envtest/volumegroup_controller_test.go:68]:
```suggestion

				err = createVolumeObjects()
				Expect(err).NotTo(HaveOccurred())

				err = createVolumeGroupObjects()
				Expect(err).NotTo(HaveOccurred())
```
for visual seperation

File: controllers/envtest/suite_test.go [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+		GRPCClient:   csiConn,
+		VGClient:     mockVolumeGroup,
+	}).SetupWithManager(mgr, driverConfig)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = (&persistentvolumeclaim.PersistentVolumeClaimReconciler{
+		Client:       mgr.GetClient(),
+		Scheme:       mgr.GetScheme(),
+		DriverConfig: driverConfig,
+		Log:          ctrl.Log.WithName("VolumeGroupContentController"),
```
Comment (ArbelNathan) [inline controllers/envtest/suite_test.go:143]:
```suggestion
		Log:          ctrl.Log.WithName("PersistentVolumeClaimController"),
```

File: controllers/envtest/suite_test.go [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+		ModifyVolumeGroupMembershipMock: func(volumeGroupId string, volumeIds []string, secrets map[string]string) (*csi.ModifyVolumeGroupMembershipResponse, error) {
+			return &csi.ModifyVolumeGroupMembershipResponse{}, nil
+		},
+	}
+
+	err = (&controllers.VolumeGroupReconciler{
+		Client:       mgr.GetClient(),
+		Scheme:       mgr.GetScheme(),
+		DriverConfig: driverConfig,
+		Log:          ctrl.Log.WithName("controllers").WithName("VolumeGroup"),
```
Comment (ArbelNathan) [inline controllers/envtest/suite_test.go:123]: any reason why its not like other reconcilers
`Log:          ctrl.Log.WithName("VolumeGroupController"),`

Comment (matancarmeli7) [reply controllers/envtest/suite_test.go:123]: I did it as we do in the [`main.go`](https://github.com/IBM/csi-volume-group-operator/blob/develop/main.go#L92-L98) file

File: controllers/envtest/volumegroupcontent_controller_test.go (hunk 2 above)
Comment (matancarmeli7) [reply controllers/envtest/volumegroupcontent_controller_test.go]: done

File: controllers/envtest/volumegroup_controller_test.go (hunk 1 above)
Comment (matancarmeli7) [reply controllers/envtest/volumegroup_controller_test.go]: done

File: controllers/envtest/volumegroup_controller_test.go (hunk 3 above)
Comment (matancarmeli7) [reply controllers/envtest/volumegroup_controller_test.go]: done

File: controllers/envtest/suite_test.go (hunk 6 above)
Comment (ArbelNathan) [reply controllers/envtest/suite_test.go:122]: ok

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**